python/lsst/ts/m2com/controller.py

```diff
@@ -496,3 +544,88 @@ async def _check_command_status(
             self.log.debug(f"Only get the acknowledgement of {command_name}.")
 
         return False
+
+    async def power(self, power_type, status, expected_state=None):
+        """Power on/off the motor/communication system.
+
+        Parameters
+        ----------
+        power_type : enum `PowerType`
+            Power type.
+        status : `bool`
+            True if turn on the power; False if turn off the power.
+        expected_state : enum `PowerSystemState` or `None`, optional
+            Expected state of the power system. This is used in the unit test
+            only. Put None in general. (the default is None)
+        """
+
+        await self.write_command_to_server(
+            "power",
+            message_details={"powerType": int(power_type), "status": status},
+        )
+        await self._check_power_status_is_expected(
+            power_type, status, expected_state=expected_state
+        )
+
+    async def _check_power_status_is_expected(
+        self, power_type, status, expected_state=None, timeout=10.0
+    ):
+        """Check the power status is expected or not.
+
+        Parameters
+        ----------
+        power_type : enum `PowerType`
+            Power type.
+        status : `bool`
+            True if turn on the power; False if turn off the power.
+        expected_state : enum `PowerSystemState` or `None`, optional
+            Expected state of the power system. This is used in the unit test
+            only. Put None in general. (the default is None)
+        timeout : `float`, optional
+            Timeout in second. (the default is 10.0)
+
+        Raises
+        ------
+        `RuntimeError`
+            When the power status is not expected.
+        """
+
+        if expected_state is None:
+            expected_state = (
+                PowerSystemState.PoweredOn
+                if status is True
+                else PowerSystemState.PoweredOff
+            )
+
+        time_wait_power_status_update = 0.5
+        time_start = time.monotonic()
+        while (time.monotonic() - time_start) < timeout:
```

Comment:
I have to admit I struggled to understand what is going on here, and I am not sure I actually understand it. 

Consider adding a "Notes" session in the docstring explaining what it is that the method does. Especially while loop and the nested if statements that `return`.

Reply:
Updated in the commit: [915bd44](https://github.com/lsst-ts/ts_m2com/pull/16/commits/915bd4406424bf702674b475f644a3b449d15726). Thanks!